File: shared/clang-wrapper/src/main/java/io/ygdrasil/koreos/clang/ClangParsingException.java
// SPDX-License-Identifier: MIT
package io.ygdrasil.koreos.clang;

/**
 * Exception thrown when parsing fails.
 */
public class ClangParsingException extends ClangException {
    public ClangParsingException(String message) {
        super(message);
    }

    public ClangParsingException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: shared/clang-wrapper/src/main/kotlin/io/ygdrasil/koreos/clang/ClangParsingException.kt
// SPDX-License-Identifier: MIT
package io.ygdrasil.koreos.clang

/**
 * Exception thrown when parsing fails.
 */
class ClangParsingException @JvmOverloads constructor(
    message: String,
    cause: Throwable? = null
) : ClangException(message, cause)
